src/pointer/invariant.rs

```diff
@@ -42,6 +42,46 @@ pub trait Aliasing: Sealed {
 pub trait Alignment: Sealed {}
 
 /// The validity invariant of a [`Ptr`][super::Ptr].
+///
+/// # Safety
+///
+/// In this section, we will use `Ptr<T, V>` as a shorthand for `Ptr<T, I:
+/// Invariants<Validity = V>>` for brevity.
+///
+/// Each `V: Validity` defines a set of bit values which may appear in the
+/// referent of a `Ptr<T, V>`. Each `V: Validity`, in its documentation,
+/// provides a definition of this set in terms of any referent type (in other
+/// words, this definition holds for all `T: ?Sized`).
+///
+/// Let `S(T, V)` be the set of bit values permitted to appear in the referent
```

Comment:
The set of things that can be in the referent of a `NonNull<T>` is different than the set of things that can be in a `T`. I think you mean the latter, not the former.